### PR TITLE
feat: Support SCAIL WanVideo model

### PR DESCRIPTION
**Pull request by kijai**
Adds model support for SCAIL: https://github.com/zai-org/SCAIL

Compatible weights:

https://huggingface.co/Kijai/WanVideo_comfy/blob/main/SCAIL/Wan21-14B-SCAIL-preview_comfy_bf16.safetensors

https://huggingface.co/Kijai/WanVideo_comfy_fp8_scaled/blob/main/SCAIL/Wan21-14B-SCAIL-preview_fp8_scaled_mixed.safetensors

Preprocessor nodes:

https://github.com/kijai/ComfyUI-SCAIL-Pose

Example [scail_native_context_test.json](https://github.com/user-attachments/files/25526490/scail_native_context_test.json):

https://github.com/user-attachments/assets/2e9203d7-d382-42ed-a567-6e89f4105421

Example pose input:

https://github.com/user-attachments/assets/2569c053-cdbf-4063-9ad2-8a9a6dd6aa86

### CODE REVIEW

File: comfy/ldm/wan/model.py [hunk 1]
```diff
@@ -1621,3 +1621,110 @@ def block_wrap(args):
...
+        x = comfy.ldm.common_dit.pad_to_patch_size(x, self.patch_size)
+
+        t_len = t
+        if time_dim_concat is not None:
+            time_dim_concat = comfy.ldm.common_dit.pad_to_patch_size(time_dim_concat, self.patch_size)
+            x = torch.cat([x, time_dim_concat], dim=2)
+            t_len = x.shape[2]
+
+        if "reference_latent" in kwargs:
+            t_len += 1
```
[the remark this answers is not in the document]
Comment (kijai) [reply comfy/ldm/wan/model.py]: It's never passed as None.

[the remark this answers is not in the document]
Comment (kijai) [reply comfy/ldm/wan/model.py]: Yes, always single frame.

File: comfy/model_base.py [hunk 2]
```diff
@@ -1466,6 +1466,59 @@ def process_timestep(self, timestep, x, denoise_mask=None, **kwargs):
...
+    def extra_conds_shapes(self, **kwargs):
+        out = {}
+        ref_latents = kwargs.get("reference_latents", None)
+        if ref_latents is not None:
+            out['reference_latent'] = list([1, 16, sum(map(lambda a: math.prod(a.size()), ref_latents)) // 16])
+
+        pose_latents = kwargs.get("pose_video_latent", None)
+        if pose_latents is not None:
+            out['pose_latents'] = pose_latents.shape
+
```
[the remark this answers is not in the document]
Comment (kijai) [reply comfy/model_base.py:1540]: Or just change 16 to 20 as the mask is always used?
